crates/sim2h/src/lib.rs

```diff
@@ -25,35 +24,69 @@ pub use crate::message_log::MESSAGE_LOGGER;
 use crate::{crypto::*, error::*};
 use cache::*;
 use connection_state::*;
-//use lib3h::rrdht_util::*;
 use lib3h_crypto_api::CryptoSystem;
 use lib3h_protocol::{
     data_types::{EntryData, FetchEntryData, GetListData, Opaque, SpaceData, StoreEntryAspectData},
     protocol::*,
     types::SpaceHash,
     uri::Lib3hUri,
 };
+use url2::prelude::*;
 
-use websocket::streams::{StreamEvent, StreamManager};
 pub use wire_message::{WireError, WireMessage};
 
+use in_stream::*;
 use log::*;
 use parking_lot::RwLock;
 use rand::{seq::SliceRandom, thread_rng};
 use std::{
     collections::{HashMap, HashSet},
     convert::TryFrom,
+    sync::Arc,
 };
 
+use holochain_locksmith::Mutex;
+
+pub(crate) trait MutexExt<T> {
+    fn f_lock(&self) -> holochain_locksmith::MutexGuard<T>;
+}
+
+impl<T> MutexExt<T> for Mutex<T> {
+    fn f_lock(&self) -> holochain_locksmith::MutexGuard<T> {
+        self.try_lock_for(std::time::Duration::from_millis(1000))
+            .expect("failed to obtain mutex lock")
```

Comment:
No need for a result oriented api here?

Reply:
This is a case where I totally prefer a hard panic! - If we haven't been able to get a mutex lock in this time frame, it probably means we have a dead-lock.